tests/System.IO.Abstractions.TestingHelpers.Tests/MockFileSystemTests.cs

```diff
@@ -351,6 +352,22 @@ public void MockFileSystem_Constructor_ThrowsForNonRootedCurrentDirectory()
             Assert.AreEqual("currentDirectory", ae.ParamName);
         }
 
+        [Test]
+        public void MockFileSystem_Constructor_ShouldExecuteFastWithLargeAmountOfFiles()
+        {
+            var filesCount = 100000;
+            var maxDirectoryDepth = 8;
+            var testData = Enumerable.Range(0, filesCount).ToDictionary(
+                i => XFS.Path(@$"C:\{string.Join(@"\", Enumerable.Range(0, i % maxDirectoryDepth + 1).Select(i => i.ToString()))}\{i}.bin"),
+                i => new MockFileData(i.ToString()));
+            var stopWatch = Stopwatch.StartNew();
+
+            var mockFileSystem = new MockFileSystem(testData);
+
+            stopWatch.Stop();
+            Assert.IsTrue(stopWatch.Elapsed < TimeSpan.FromSeconds(30));
```

Comment:
Great that you wrote a test to verify your changes! 

I prefer to keep the tests on the functional level and very fast. Would you mind moving this test case to the `benchmarks` directory to accommodate for this?

Reply:
I moved the test case to the `benchmarks` project. I had to reference `System.IO.Abstractions.TestingHelpers.csproj`.